library/certs.c

```diff
@@ -220,24 +220,24 @@ const size_t mbedtls_test_ca_pwd_rsa_len = sizeof( mbedtls_test_ca_pwd_rsa ) - 1
 
 const char mbedtls_test_srv_crt_rsa[] =
```

Comment:
Please follow same logic we have for the [test CA](https://github.com/ARMmbed/mbedtls/blob/development/library/certs.c#L120-L179)

Reply:
> Thank you for your contribution!

No problem :)

> In addition, it would be appreciated if you add an entry to our ChangeLog

Of course, will do!

> Please follow same logic we have for the test CA

You want me to refactor it? I can do that no problem. Do you mean putting the new certificate string in a macro or only using the new certificate when `TEST_CA_CRT_RSA_SHA256` is defined?

Reply:
I meant having similar logic like in the CA:
defining `TEST_CRT_RSA_SHA256` as the certificate if `MBEDTLS_SHA256_C` is defined, and then defining `TEST_CRT_RSA_SOME`. After that define `TEST_CRT_RSA_SHA1` if `TEST_CRT_RSA_SOME` not defined, or `MBEDTLS_SHA1_C` is defined, and set `mbedtls_test_srv_crt_rsa[]` to be  `TEST_CRT_RSA_SHA1` if `TEST_CRT_RSA_SOME`  not defined.

Note we will probably also ask you to backport your fix to our Long Time Support branches ( `mbedtls-2.1` and `mbedtls-2.7` ), but I recommend you do it only after this PR will be finally approved, to avoid multiple work.

Reply:
@RonEld okay, that's what I thought that you meant. I will do this as soon as I get back home (in a few hours).

Okay, will gladly do it :)